pyHalo/realization_extensions.py

```diff
@@ -425,8 +438,15 @@ def add_primordial_black_holes(self, pbh_mass_fraction, kwargs_pbh_mass_function
                                           self._realization.lens_cosmo, kwargs_realization=self._realization._prof_params)
 
         kwargs_pbh_mass_function['mass_fraction'] = mass_fraction_clumpy
-        realization_with_clustering = self.add_correlated_structure(kwargs_pbh_mass_function, mass_definition, x_image_interp_list, y_image_interp_list,
-                                                        r_max_arcsec, arcsec_per_pixel, rescale_normalizations)
+        if combine_inds == None:
+            realization_with_clustering = self.add_correlated_structure(kwargs_pbh_mass_function, mass_definition, x_image_interp_list, y_image_interp_list,
+                                                            r_max_arcsec[0], arcsec_per_pixel, rescale_normalizations)
+        if combine_inds != None:
+            realization_with_clustering_A = self.add_correlated_structure(kwargs_pbh_mass_function, mass_definition, x_image_interp_list[0:2], y_image_interp_list[0:2],
```

Comment:
You should go through the x_image_interp_list and y_image_interp_list one at a time, and join the realizations together in each iteration of the for loop

Reply:
Yeah, that probably wouldn't be too much slower